pkg/controller/instance/controller_reconcile.go

```diff
@@ -123,193 +184,100 @@ func (igr *instanceGraphReconciler) reconcileInstance(ctx context.Context) error
 		igr.state.ResourceStates[resourceID] = &ResourceState{State: ResourceStatePending}
 	}
 
-	// Reconcile resources in topological order
-	for _, resourceID := range igr.runtime.TopologicalOrder() {
-		if err := igr.reconcileResource(ctx, resourceID); err != nil {
-			return err
-		}
-
-		// Synchronize runtime state after each resource
-		if _, err := igr.runtime.Synchronize(); err != nil {
-			return fmt.Errorf("failed to synchronize reconciling resource %s: %w", resourceID, err)
-		}
+	config := applyset.Config{
+		ToolLabels:         igr.instanceSubResourcesLabeler.Labels(),
+		FieldManager:       FieldManagerForApplyset,
+		ToolingID:          KROTooling,
+		LoadCallback:       igr.processLoad,
+		AfterApplyCallback: igr.processAfterApply,
+		Log:                igr.log,
 	}
 
-	return nil
-}
-
-// setupInstance prepares an instance for reconciliation by setting up necessary
-// labels and managed state.
-func (igr *instanceGraphReconciler) setupInstance(ctx context.Context, instance *unstructured.Unstructured) error {
-	patched, err := igr.setManaged(ctx, instance, instance.GetUID())
+	aset, err := applyset.New(instance, igr.restMapper, igr.client, config)
```

Comment:
nit: im wondering if you need to pass in instance here because of https://github.com/kro-run/kro/pull/561/files#diff-3e23972797bd9d87e2581d861789fbe1816800bf43794f45ae4d46f7a304e790R115 since you have the gvr with igr.gvr, you may want to use that as a parent instead?

Reply:
Extracting  PartialObjectMetadata in New() now. We need the parent name and namespace as well.